components/scream/cmake/SetCompilerFlags.cmake

```diff
@@ -151,6 +155,9 @@ ELSE ()
   IF (DEBUG_FFLAGS)
     SET (CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS_DEBUG} ${DEBUG_FFLAGS}")
   ELSE ()
+    IF ("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "Intel")
+      SET(CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS_DEBUG} -no-vec -fp-model strict")
```

Comment:
I don't think we want -fp-model to track whether the build is debug; rather, we want SCREAM_FPMODEL to control the floating point rules, independent of optimization level. The reason is that we want to be able to do BFB testing (which requires control of the floating point model) at high optimization (whose various manipulations are still enabled, except for those that are FP-changing), if desired.

As a matter of taste, SCREAM_FPMODEL could be auto-set to ON if the build is debug, although we don't do that in HOMMEXX.